### PR TITLE
Add liveness check framework

### PR DESCRIPTION
**Pull request by johngmyers**

### CODE REVIEW

File: reporting/src/test/java/com/proofpoint/reporting/TestHealthBean.java [hunk 1]
```diff
@@ -97,7 +148,7 @@ public String getString()
 
     @Test(expectedExceptions = RuntimeException.class,
             expectedExceptionsMessageRegExp = ".*field cannot have both @HealthCheck and @HealthCheckRemoveFromRotation annotations")
-    public void testFieldBothAnnotatins()
+    public void testFieldRegularAndRemoveAnnotatins()
```
Comment (gwittel) [inline reporting/src/test/java/com/proofpoint/reporting/TestHealthBean.java]: annotations

File: reporting/src/test/java/com/proofpoint/reporting/TestHealthBean.java [hunk 2]
```diff
@@ -81,9 +82,59 @@ public void testFieldNotAtomicReference()
...
+            public String getString()
+            {
+                return null;
+            }
+        });
+    }
+
+    @Test(expectedExceptions = RuntimeException.class,
+            expectedExceptionsMessageRegExp = ".*field cannot have both @HealthCheck and @HealthCheckRestartDesired annotations")
+    public void testFieldRegularAndRestartAnnotatins()
```
Comment (gwittel) [inline reporting/src/test/java/com/proofpoint/reporting/TestHealthBean.java]: annotations

File: reporting/src/test/java/com/proofpoint/reporting/TestHealthBean.java [hunk 3]
```diff
@@ -81,9 +82,59 @@ public void testFieldNotAtomicReference()
...
+            public String getString()
+            {
+                return null;
+            }
+        });
+    }
+
+    @Test(expectedExceptions = RuntimeException.class,
+            expectedExceptionsMessageRegExp = ".*field cannot have both @HealthCheckRemoveFromRotation and @HealthCheckRestartDesired annotations")
+    public void testFieldRestartAndRemoveAnnotatins()
```
Comment (gwittel) [inline reporting/src/test/java/com/proofpoint/reporting/TestHealthBean.java]: annotations

**Comment by johngmyers** [comment]
Fixed and merged

**Pull request closed without merging**